pkg/sfu/buffer/buffer.go

```diff
@@ -68,20 +68,13 @@ type Buffer struct {
 	lastPacketRead int
 	bitrate        atomic.Value
 	bitrateHelper  [4]int64
-	lastSRNTPTime  NtpTime
-	lastSRRTPTime  uint32
-	lastSRRecv     int64 // Represents wall clock of the most recent sender report arrival
-	lastTransit    uint32
 
 	pliThrottle int64
-	lastPli     int64
 
-	started    bool
-	stats      StreamStats
-	rrSnapshot *receiverReportSnapshot
-
-	highestSN uint16
-	cycle     uint16
+	rtpStats                    *RTPStats
+	rrSnapshotId                uint32
+	rembSnapshotId              uint32
+	connectionQualitySnapshotId uint32
```

Comment:
Moving all stats stuff into `RTPStats`.

Also introducing notion of a snapshot  in `RTPStats` module. Will add more comments there. Basically, users of `RTPStats` can create a snap shot which returns an id. They can then use that id to get data from `RTPStats` module using that snap shot id as the base.

Here we are using it for
- RTCP Receiver Report
- REMB (we do not use it, but the code is there). So, creating a snap shot means a bit of extra work for this snap shot in `RTPStats` module. Snapshot is very lightweight now, but worth keeping an eye on it.
- For connection quality stats

As these can all run on different tickers, just keeping snap shots and operating on them gives the flexibility.